datastore/google/cloud/datastore/_gax.py

```diff
@@ -71,6 +73,13 @@ def _grpc_catch_rendezvous():
     """
     try:
         yield
+    except GaxError as exc:
+        error_code = exc_to_code(exc.cause)
+        error_class = _GRPC_ERROR_MAPPING.get(error_code)
+        if error_class is None:
+            raise
+        else:
+            raise error_class(exc.cause.details())
```

Comment:
Assuming that it works like I claim it does, can this be changed to use `six.reraise` and maintain the _original_ traceback, please? I have done the same in [this PR][1] on GAX, pending some real-world testing. Assuming I am right about exactly how this works, it should make debugging substantially easier.

  [1]: https://github.com/googleapis/gax-python/pull/168/files#diff-5ce6a350c255795b7c02e598ab053273R149